_inc/connect-button.js

```diff
@@ -54,12 +56,26 @@ jQuery( document ).ready( function( $ ) {
 				},
 				error: jetpackConnectButton.handleConnectionError,
 				success: function( data ) {
+					jetpackConnectButton.fetchPlanType();
 					window.addEventListener( 'message', jetpackConnectButton.receiveData );
 					jetpackConnectIframe.attr( 'src', data.authorizeUrl );
 					$( '.jp-connect-full__button-container' ).html( jetpackConnectIframe );
 				},
 			} );
 		},
+		fetchPlanType: function() {
```

Comment:
Just curious, wouldn't it be easier to fetch the plan in PHP with `Jetpack_Plan::get()` and pass it to the file with `wp_localize_script`?

Reply:
we tried, we don't have a blog token at that point yet, we can't make the request :)

Reply:
@jeherve I'm not sure if we're allowed to do so. Anything that we're passing to the script using `wp_localize_script` happens before the user clicks the "Set up Jetpack" button, i.e. before they agree to terms and conditions. In current implementation, we're requesting that data after the button has been clicked.